General/ProgrammaticCodeManipulation.md

```diff
@@ -0,0 +1,136 @@
+# Programmatic Code Manipulation (Pharo 11)
+
+In this article, we will present some code snippets that demonstrate how we can manipulate the source code (packages, classes, methods, variables, traits, etc.) programmatically, without using the SystemBrowser. Those techniques can be very useful for reflective operations, code generation, testing, etc.
+
+This article explains only those operations that **modify** the code. We will write a separate article on **querrying code** (looking for method senders, class references, pragma usages, etc.).
+
+## Packages
+
+### Creating a new package
+
+```st
+package := self packageOrganizer ensurePackage: 'PackageName'.
+```
+
+In Playground, you can use any object or class instead of `self`:
+
+```st
+package := Object packageOrganizer ensurePackage: 'PackageName'.
+```
+
+### Removing a package
+
+```st
+package removeFromSystem.
+```
+
+**Be careful!** In Pharo 11 and earlier, this will not properly remove classes from the system. So make sure to remove all classes before you remove the package.
+
+### Creating a package without installing it
+
+_(how to create an instance of a package without installing it into the image? - e.g., for testing)_
+
+## Classes
+
+### Creating a new class
+
+To create a new class, you can use the same code that appears in SystemBrowser. Here is an example with **Fluid** class definition.
+
+```st
+builder := Object << #MyClass
	slots: { };
	package: 'MyPackage'.
	
class := builder install.
+```
+
+The first statement will return an instance of `FluidClassBuilder`. The second statement will build the class, install it into the image, and assign it to the variable. If `MyPackage` does not exist, it will be created.
+
+### Creating a new class without installing it
+
+Here is an example of how you can create a new class without installing it into the system. You can do that by sending `build` message instead of `install` as in the previous example.
+
+```st
+builder := OrderedCollection << #MyCollection.
myCollectionClass := builder build.

myCollectionClass compile: 'generate: aNumber
    "Generate aNumber random integers"
    | rand |
    rand := Random new.
    aNumber timesRepeat: [ self add: (rand nextInteger: 100) ]'.
	
myCollection := myCollectionClass new.
myCollection generate: 4. "a MyCollection(60 5 40 11)"
```

Comment:
This is weird on Github I don't see the lines returns well. I hope it's just a display problem.

![image](https://github.com/pharo-open-documentation/pharo-wiki/assets/9519971/6a9bc8b0-1eb0-4320-9e96-aafeced7a678)